src/ngraph/runtime/interpreter/CMakeLists.txt

```diff
@@ -14,20 +14,19 @@
 # limitations under the License.
 # ******************************************************************************
 
-if(NGRAPH_DISTRIBUTED_ENABLE)
-    find_package(MPI REQUIRED)
-    add_definitions(-DNGRAPH_DISTRIBUTED)
-    include_directories(SYSTEM ${MPI_C_INCLUDE_PATH} ${MPI_CXX_INCLUDE_PATH})
-    link_directories(${MPI_C_LIBRARIES} ${MPI_CXX_LIBRARIES})
-    link_libraries(${MPI_CXX_LIBRARIES})
-endif()
-
 if (NGRAPH_INTERPRETER_ENABLE)
     add_library(interpreter_backend SHARED int_backend.cpp)
     set_target_properties(interpreter_backend PROPERTIES VERSION ${NGRAPH_VERSION} SOVERSION ${NGRAPH_API_VERSION})
     target_link_libraries(interpreter_backend PUBLIC ngraph)
     set_target_properties(interpreter_backend PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${NGRAPH_BUILD_DIR})
 
+if(NGRAPH_DISTRIBUTED_ENABLE)
+    add_definitions(-DNGRAPH_DISTRIBUTED)
+    add_dependencies(interpreter_backend libmlsl)
+    target_include_directories(interpreter_backend SYSTEM PUBLIC libmlsl)
```

Comment:
If we do decide that PRIVATE works then make sure you catch all cases.

Reply:
Changed to PRIVATE